### ISSUE TITLE
Documentation/Implementation for output timing data

### ISSUE THREAD

**Issue by mrhardman** [mabarnes/moment_kinetics#84]
Currently the code has a TimerOutput depency in moment_kinetics.jl, however, by default timing data does not seem to be output. 

An enhancement therefore is to document or implement the output of timing data.

**Comment by johnomotani** [comment]
Use of TimerOutputs is controlled by `run_type`, which is a `const` variable in `moment_kinetics_input.jl`:
https://github.com/mabarnes/moment_kinetics/blob/e01ed55495ac3bbde7bba8c93d9c667e12438ef6/src/moment_kinetics_input.jl#L17
Set `run_type = performance_test` and some timing output should be printed. Note the timing that is displayed is just from the rank-0 process.

Agree documenting would be useful!

Also would be nice to be able to set `run_type` from the input file (possibly giving it a better name...). [Digression: I'm not sure `run_type = scan` is really useful any more. It was an early attempt at making input more flexible to allow parameter scans, but now each run can use multiple processes, and runs can be launched by passing a `Dict` that can be modified to run a parameter scan from a script/function (as is done by the tests).]

**Comment by mrhardman** [comment]
Is there a reason for why we do not always print timing info by default, as in other codes? Did you expect timing data to be expensive to collect in Julia?

**Comment by mrhardman** [comment]
After reviewing the code and testing the current functionality, it seems like the available timing packages in Julia (e.g. https://juliapackages.com/p/timeroutputs) are designed for benchmarking individual functions rather than simply reporting the total execution time of the code. I am more interested in the latter information, to make a rough estimate how long to schedule jobs to run for on HPC clusters. I think I can satisfy this more simply with the $SECONDS variable in my bash scripts. 

I note that there appear to be issues with getting good timings from Julia. The reported time from the "performance_test" option was very different to the actual run time on the machine, likely due to the pre-compilation overhead. Whilst we are running uncompiled code in development, we do care about this subtlety!

I'll this this issue open as I think we can still improve this feature.

**Comment by johnomotani** [comment]
Oops, sorry the close/reopen was just a fat finger hitting the mouse!

Agree it's tricky, don't know of an easy way to time stuff without counting (pre-)compilation time.

I don't really remember, but the reason I turned off timing by default could have been that it was noisy when I was running parameter scans, and/or that at the time run time was much less than compilation time, so to get information that was any use, the 'performance_test' would do the run twice, only timing the second one (but that's obviously inefficient for 'normal' runs.

For 'normal output' we could easily do something like timing from the end of the first timestep (which would skip all the compilation and setup time) and print/save that. We could also try timing the setup: I'm not sure but if we start/stop the timer inside `setup_moment_kinetics()`, hopefully all the functions called inside get compiled when `setup_moment_kinetics()` gets compiled, so we would (might?) get a sensible timing of the allocation/initialisation. At worst I guess we'd get an inconsistent timing that might or might not include some compilation time along with the setup time.

When I've been testing/debugging the 1D1V runs, I've taken to printing the current time along with the number of steps when output is written, as that allows me to check if some step is taking longer than the others (which probably means the code has crashed).

**Comment by johnomotani** [comment]
Yes, I'm on 1.8. I did have some trouble getting it running on Marconi, ended up compiling from source - and to get that to work had to compile `binutils` from source to get a more recent version of `ld` - but it seems to work.

For writing time, I did something like
```
...

using Dates

...

    println("beginning time advance...", Dates.format(now(), dateformat"H:MM:SS"))
```
(see https://github.com/mabarnes/moment_kinetics/commit/a38a0c39ae6b66f929f671c5a5ed3c6fb0fa3423 from #76) and yes, just eyeball the differences.
